### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by azmeuk**
This patch:
- Adds a `DNSServer` class in `dnserver.py`. It has a simple interface like `server.start()`, `server.stop()` and `server.is_running`. I hope this can be used as a dns server fixture in other projects.
- Adds a tox configuration for currently active python versions.
- Adds a simple unit test, that just makes a request to the DNS.
- Adds a GHA configuration, for safety.
- Requires dnslib `0.9.20`
- It refactors the code so it uses zone strings instead of zone path, so it is easier to use in a python context.

@samuelcolvin

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Hi really happy to get this to a point where it can be released to pypi, including unit tests.

I would prefer not to use tox, but rather rely on github actions to test on multiple versions - this is mostly because it's the workflow I know best and can therefore support most easily.

Could you refactor this to follow the example of [dirty_equals](https://github.com/samuelcolvin/dirty-equals)? If there are bits you find confusing, feel free to leave a comment on this PR and omit them.

**Comment by azmeuk** [comment]
I removed tox and adapted the GHA yaml config file following what I could read in dirty_equal. I am curious how do you test several python versions on your development machine without tox?

At first I intended to plug pre-commit and some style tools like black, but I thought this would be better in another PR.

**Comment by samuelcolvin** [comment]
> I removed tox and adapted the GHA yaml config file following what I could read in dirty_equal. I am curious how do you test several python versions on your development machine without tox?

I have multiple virtualenvs, and if I really need to test between versions quickly, I activate each in a different terminal; but mostly it's not necessary.

> At first I intended to plug pre-commit and some style tools like black, but I thought this would be better in another PR.

Sounds good.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
```
Comment (samuelcolvin) [inline .github/workflows/ci.yml:1]: not sure what this is doing, but I think it shouldn't be there.

Comment (azmeuk) [reply .github/workflows/ci.yml:1]: yamllint [yells when this is absent](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.document_start)

Comment (samuelcolvin) [reply .github/workflows/ci.yml:1]: I'll merge and check what github thinks, it definitely doesn't have to be there.

**Comment by samuelcolvin** [comment]
thanks so much, I'll make further changes in a new PR.

**Pull request merged**